sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1845,18 +1846,21 @@ public PCollection<KafkaRecord<K, V>> expand(PBegin input) {
                 "Offsets committed due to usage of commitOffsetsInFinalize() and may not capture all work processed due to use of withRedistribute() with duplicates enabled");
           }
 
-          if (kafkaRead.getRedistributeNumKeys() == 0) {
```

Comment:
I think the names maybe need to be the same for update compatability. Can we just leave them as they were?